crates/bevy_winit/src/lib.rs

```diff
@@ -45,7 +84,23 @@ pub fn winit_runner(mut app: App) {
     );
 
     log::debug!("Entering winit event loop");
-    event_loop.run(move |event, event_loop, control_flow| {
+
+    let should_return_from_run = match app.resources.get::<WinitConfig>() {
+        Some(config) => {
+            cfg!(any(
+                target_os = "windows",
+                target_os = "macos",
+                target_os = "linux",
+                target_os = "dragonfly",
+                target_os = "freebsd",
+                target_os = "netbsd",
+                target_os = "openbsd"
+            )) && config.return_from_run
+        }
+        _ => false,
+    };
+
+    let event_handler = make_event_handler(move |event, event_loop, control_flow| {
```

Comment:
`make_event_handler` is used here because I couldn't find a type signature that pleased the compiler in order to put this closure in a local variable like this. If anyone knows a cleaner way I'm happy to learn.

Reply:
Thanks! Done.